cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/PreventBrokerScaleDownOperations.java

```diff
@@ -0,0 +1,156 @@
+
+        Promise<Set<Integer>> cannotScaleDown = Promise.promise();
+        ReconcilerUtils.clientSecrets(reconciliation, secretOperator)
+                .compose(compositeFuture -> {
+                    Promise<Void> resultPromise = Promise.promise();
+
+                    final Future<Collection<TopicDescription>> descriptions;
+                    try {
+                        String bootstrapHostname = KafkaResources.bootstrapServiceName(reconciliation.name()) + "." + reconciliation.namespace() + ".svc:" + KafkaCluster.REPLICATION_PORT;
+                        LOGGER.debugCr(reconciliation, "Creating AdminClient for cluster {}/{}", reconciliation.namespace());
+                        Admin kafkaAdmin = adminClientProvider.createAdminClient(bootstrapHostname, compositeFuture.resultAt(0), compositeFuture.resultAt(1), "cluster-operator");
+
+                        Future<Set<String>> topicNames = topicNames(reconciliation, kafkaAdmin);
+                        descriptions = topicNames.compose(names -> {
+                            LOGGER.infoCr(reconciliation, "Topic names {}", names);
+                            return describeTopics(reconciliation, kafkaAdmin, names);
+                        });
+                        descriptions
+                                .compose(topicDescriptions -> {
+                                    LOGGER.debugCr(reconciliation, "Got {} topic descriptions", topicDescriptions.size());
+                                    Set<Integer> idsContainingPartitionReplicas = new HashSet<>();
+                                    // Provides the node IDs which the user would like to remove first
+                                    for (Integer id: idsToBeRemoved)   {
+                                        boolean result =  brokerHasAnyReplicas(reconciliation, topicDescriptions, id);
+                                        if (result) {
+                                            idsContainingPartitionReplicas.add(id);
+                                        }
+                                    }
+                                    cannotScaleDown.complete(idsContainingPartitionReplicas);
+                                    kafkaAdmin.close();
+                                    return cannotScaleDown.future();
+                                }).recover(error -> {
+                                    LOGGER.warnCr(reconciliation, "failed to get topic descriptions", error);
+                                    cannotScaleDown.fail(error);
+                                    kafkaAdmin.close();
+                                    return Future.failedFuture(error);
+                                });
+
+                    } catch (KafkaException e) {
+                        LOGGER.warnCr(reconciliation, "Kafka exception while trying to check if broker contains partition replicas", e.getMessage());
```

Comment:
`Failed to check if broker contains any partition replicas` would be better maybe? Also, should you add the broker ID?